albumentations/augmentations/transforms.py

```diff
@@ -2706,7 +2706,7 @@ def get_params_dependent_on_targets(self, params):
             shape = [c]
```

Comment:
This comment is not related to the bugfix itself. It points to the possible bug in implementation, when per_channel = True. In this case, the condition check will set shape to `shape = [c]`, which I read as "apply same noise vector of [c] channels to all pixels". Probably, it should be `shape = [h,w,1]` to apply image-level noise to all channels instead.

Reply:
@BloodAxe
I a little bit confused what you mean.
Do you mean that in your mind condition `per_channel` means that we need to generate noise with shape `[h, w]`?
In my mind condition `per_channel=True` means that we will use `c` equal to image `c` shape when we produce noise tensor, else we use `c=1`.
If `elementwise=True` we set `[h, w]` of noise vector equal to image `[h, w]`. If `elementwise=False` noise vector `h = w = 1`